go/test/endtoend/cluster/vttablet_process.go

```diff
@@ -352,7 +354,7 @@ func (vttablet *VttabletProcess) WaitForBinLogPlayerCount(expectedCount int) err
 
 // WaitForBinlogServerState wait for the tablet's binlog server to be in the provided state.
 func (vttablet *VttabletProcess) WaitForBinlogServerState(expectedStatus string) error {
-	timeout := time.Now().Add(10 * time.Second)
+	timeout := time.Now().Add(vttabletStateTimeout)
```

Comment:
`time.Now()` calls are considered to be expensive. I believe the idiomatic way of  running this kind of code, while we're fixing things, is to have a `context.WithTimeout`

The entire code would look like this:
```go
	ctx, cancel := context.WithTimeout(vttabletStateTimeout)
	defer cancel()
	t := time.NewTicker(300 * time.Millisecond)
	defer t.Stop()
	for {
		if vttablet.getVarValue("UpdateStreamState") == expectedStatus {
			return nil
		}
		select {
		case err := <-vttablet.exit:
			return fmt.Errorf("process '%s' exited prematurely (err: %s)", vttablet.Name, err)
		case <- ctx.Done():
			return fmt.Errorf("vttablet %s, expected status not reached", vttablet.TabletPath)
		case <- t.C:
		}
	}
```

Reply:
Thanks, updated with suggested logic